src/shell.c

```diff
@@ -45,21 +46,46 @@ void exec(char *command, char *args[16], int argc) {
 int main() {
     char input[MAX_SIZE];
     char command[MAX_SIZE], *args[16];
+    char cwd[MAX_SIZE];
+
+    char *username = getlogin();
+    char hostname[MAX_SIZE];
+    gethostname(hostname, sizeof(hostname));
+
+    getcwd(cwd, sizeof(cwd));
+    printf("%s@%s:%s$ ", username, hostname, cwd);
     while (1) {
         fgets(input, MAX_SIZE, stdin);
         // ignores the case when the input is empty
         if (strcmp(input, "\n") == 0) {
+            printf("%s@%s:%s$ ", username, hostname, cwd);
             continue;
         }
 
         // removes "\n" from end of the line
         input[strlen(input) - 1] = '\0';
 
         int argc = parse(input, command, args);
-        if (vfork() == 0) {
+
+        pid_t pid = fork();
```

Comment:
Make it a vfork instead of a fork, so that you don't have to use the wifexited and wait logic. (In vfork, the parent automatically waites for the child process to finish).